packages/snaps-cli/src/config.ts

```diff
@@ -562,11 +562,7 @@ export const SnapsWebpackConfigStruct = object({
     {},
   ),
 
-  environment: defaulted(record(string(), unknown()), {
-    NODE_DEBUG: false,
-    NODE_ENV: 'production',
-    DEBUG: false,
-  }),
+  environment: defaulted(record(string(), unknown()), {}),
```

Comment:
Previously, the defaults would be completely ignored if any value was used. I've changed it to merge the defaults with the custom values provided in the config.